agent/app/agent.go

```diff
@@ -579,3 +585,26 @@ func (agent *ecsAgent) startACSSession(
 	seelog.Critical("ACS Session handler should never exit")
 	return exitcodes.ExitError
 }
+
+// validateRequiredVersion validates docker version.
+// Minimum docker version supported is 1.9.0, maps to api version 1.21
+// see https://docs.docker.com/develop/sdk/#api-version-matrix
+func (agent *ecsAgent) verifyRequiredDockerVersion() (int, bool) {
+	supportedVersions := agent.dockerClient.SupportedVersions()
+	if len(supportedVersions) == 0 {
+		seelog.Critical("Could not get supported docker versions.")
+		return exitcodes.ExitError, false
```

Comment:
With the new api version change(minimum is 1.21), you probably only need to return `exitcodes.ExitTerminal` here, and then you can remove the rest of the code.

Reply:
undid this change, CP backend might still need >= logic on the api versions. Keeping exitError here.